### PR TITLE
load_store emitters optimize and applied to interpolate

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *optimize interpolate by apply load store emitters*
 - *emitters optimization to boost tails process on avx512*

### Tickets:
 - *57325*

oneDNN PR:https://github.com/openvinotoolkit/oneDNN/pull/97

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang Is the PR ready for review?

**Comment by chenhu-wang** [comment]
> @chenhu-wang Is the PR ready for review?

@dmitry-gorokhov , 57325 is ready for review. For 55387, still in process, I am ok to create separate PR for 55387.

**Comment by dmitry-gorokhov** [comment]
@EgorDuplensky please review the PR

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp [hunk 1]
```diff
@@ -441,29 +461,23 @@ struct jit_uni_interpolate_kernel_f32 : public jit_uni_interpolate_kernel, publi
...
             L(nn_loop_end_label);
 
-            step = 1;
-            L(nn_tail_loop_label);
-            {
-                cmp(reg_work_amount, 1);
-                jl(nn_tail_loop_end_label, T_NEAR);
-
-                load_scalar(xmm_val, ptr[reg_src_aux], jcp_.src_dt);
+            int tail_num = jcp_.C - (jcp_.C / step) * step;
```
Comment (EgorDuplensky) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp:464]: Can we just use % instead?

File: inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp [hunk 2]
```diff
@@ -1335,21 +1348,21 @@ struct jit_uni_interpolate_kernel_f32 : public jit_uni_interpolate_kernel, publi
 
     // always gather to Vmm, compute with Vmm, store with Xmm if scalar
     inline void gather_i32_indices(Vmm vmm_src, const Xbyak::Reg64 &base, int offset, Vmm vmm_indices, int scale,
-                                memory::data_type src_dt, bool is_scalar) {
+                                memory::data_type src_prc, bool is_scalar) {
```
Comment (EgorDuplensky) [inline inference-engine/src/mkldnn_plugin/nodes/mkldnn_interpolate_node.cpp:1351]: As I can see the idea is to get rid of oneDNN's memory::data_type* usage?
If so I propose to apply it for this gather_i32_indices as well.
I mean to change the argument type to Precision and update the calls.

**Review by EgorDuplensky — COMMENTED** [review]
No major comments from my side.

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang Do we still need smt from this PR?

**Pull request closed without merging**